hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java

```diff
@@ -96,4 +151,16 @@ private boolean isValidStakingSentinel(
             return false;
         }
     }
+
+    private boolean isValidStakingIdForCreation(
+            final String stakedIdKind, final AccountID stakedAccountId, final Long stakedNodeId) {
+        if (stakedIdKind.equals("STAKED_ACCOUNT_ID")) {
+            // current checking only account num since shard and realm are 0.0
+            return requireNonNull(stakedAccountId).accountNum() > 0;
+        } else if (stakedIdKind.equals("STAKED_NODE_ID")) {
+            return requireNonNull(stakedNodeId).longValue() >= 0;
```

Comment:
Aren't staked node IDs negative?

Reply:
No we changed to have two fields for stakedAccountId and stakedNodeId in Account now

Reply:
perfect, thanks!